src/layoutBuilder.js

```diff
@@ -388,75 +495,147 @@ function convertWordsToCodepoints(lineAsArrayOfCodepoints) {
-			// styling for each word before it was transformed and apply it to the
-			// transformed word.
-
-			// TODO I don't know how this will stack up again LTR words. Will need
-			// to evaluate once we have mixed fonts support.
-			var oldInline = inlinesBeforeTransformation[inlinesBeforeTransformation.length - 1 - index];
-			var newInline = inline;
-			if (oldInline) {
-				newInline.style = oldInline.style;
-				newInline.background = oldInline.background;
-				newInline.font = oldInline.font;
-				newInline.decoration = oldInline.decoration;
-				newInline.decorationColor = oldInline.decorationColor;
-			}
-			return line.addInline(newInline);
-		});
+/**
+ * Takes the inlines from the current line. Reorders the inlines using BIDI. After reordering
+ * the inlines it uses them to make a new line and returns that line
+ * @param {Line} line The original line that contains some inline RTL
+ * @param {StyleContextStack} styleStack The style context used when rebuilding the inlines
+ * @param {TextTools} textTools Used to rebuild the inlines
+ * @param {Object} textNode The original textNode that contains the nested inline rtl text
+ * @param {Number} availableWidth the line width being used
+ * @returns {Line}
+ */
+function addLineWithInlineRTL(
+	line,
+	styleStack,
+	textTools,
+	textNode,
+	availableWidth
+) {
+	// Turn the inlines into a single string
+	const lineElementsAsString = line.inlines.map((e) => e.text).join('');
+
+	// Use that string to create a bidi instance
+	const bidiString = bidi.from_string(lineElementsAsString).reorder_visually();
+
+	// The bidi instance contains string_arr, which is the original string represented
+	// by a single array of codepoints. Call convertWordsToCodepoints in order to
+	// get back an array (representing the line) of arrays (representing the words) of
+	// codepoints (representing the individual characters)
+	const wordsAsArraysOfCodePoints = convertWordsToCodepoints(
+		bidiString.string_arr
+	);
+	const arrayOfTransformedWords = [];
+
+	// Copied from transformLineForRtl, this fills our arrayOfTransformedWords
+	wordsAsArraysOfCodePoints.forEach((wordAsCodePoints) => {
+		const wordAsString = spreadify(
+			String.fromCharCode,
+			String
+		)(wordAsCodePoints);
+		const bidiWord = bidi.from_string(wordAsString).reorder_visually();
+		arrayOfTransformedWords.push(bidiWord.toString());
+	});
+
+	// Using this arrayOfTransformedWords, we need to rebuild the inlines,
+	// passing through the textNode original style
+	const updatedInlines = textTools.buildInlines(
+		[{ text: arrayOfTransformedWords.join(''), style: textNode.style }],
+		styleStack
+	);
+
+	// Unlike transformLineForRtl we need to build a new line here and then
+	// add each inline in turn. We know it will fit, because that check has been
+	// carried out already by buildNextLine.
+	const newLine = new Line(availableWidth);
+	updatedInlines.items.forEach((newInline) => newLine.addInline(newInline));
+	return newLine;
```

Comment:
This is the new function - almost completely the same as the old one, only change is the final step.

Reply:
This has actually changed a fair bit - it needed a lookup to allow the styles to be mapped correctly to the words in the line (for inline fonts)